devimint/src/util.rs

```diff
@@ -236,6 +236,39 @@ macro_rules! cmd {
 
 const POLL_INTERVAL: Duration = Duration::from_millis(200);
 
+/// Will retry calling `f` until it returns `Ok(true)` or `retries` times.
+/// A notable difference from [`poll`] is that `f` may fail with an error at any
+/// time and we will still keep retrying.
+pub async fn poll_max_retries<Fut>(name: &str, retries: usize, f: impl Fn() -> Fut) -> Result<()>
+where
+    Fut: Future<Output = Result<bool>>,
+{
+    let mut i = 0;
+    loop {
+        let result = f().await;
+        if i == retries - 1 {
+            match result {
+                Ok(true) => return Ok(()),
+                Ok(false) => {
+                    bail!("{name} failed to reach good state after {retries} retries");
+                }
+                Err(e) => {
+                    bail!("{name} failed after {retries} retries with: {e:?}");
+                }
+            }
+        } else {
+            match result {
+                Ok(true) => return Ok(()),
+                other => {
+                    i += 1;
+                    debug!("polling {name} failed with: {other:?}, will retry... ({i}/{retries})");
+                    task::sleep(Duration::from_secs(1)).await;
+                }
+            }
+        }
+    }
+}
```

Comment:
Feels like `poll_value` could now be implemented as a special case of this fn, but good enough.

Reply:
I think we could create a `poll_max_retries_value`. But the behavior regarding errors on the `pool` ones is a bit different. They fail immediately, so there is no "retry" on errors, it's just a simple "polling". I didn't analyze the current calls enough to see if it makes sense to change that. @Maan2003 do you have opinions here?

Reply:
I was playing around with a few solutions but seems too complicated indeed, nevermind.

Reply:
We can make it explicitly using `ControlFlow<Result<T>>` instead of `Result<Option<T>>`